compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -81,9 +68,15 @@ struct BinderLevel {
 
 struct SymbolMangler<'tcx> {
     tcx: TyCtxt<'tcx>,
-    compress: Option<Box<CompressionCaches<'tcx>>>,
     binders: Vec<BinderLevel>,
     out: String,
+
+    /// The length of the prefix in `out` (e.g. 2 for `_R`).
+    start_offset: usize,
+    /// The values are start positions in `out`, in bytes.
```

Comment:
Comment was not doc comment because it applies to all 3 fields.
Also, the names of the fields should've probably been adjusted (or the separate `struct` kept), since now they've lost a bit of context.

Reply:
Ack. Will keep this in mind when making further changes there.

I was looking into the idea of using only shortening back-references, but differences seems to be negligible. For cargo, total symbol size reduction of -0.016% (`nm target/debug/cargo | grep -o '_R.*' | wc --bytes`). Could be more useful with const generics, I suppose.